include/aos/common/tools/string.hpp

```diff
@@ -306,10 +348,17 @@ class String : public Array<char> {
         Clear();
 
         for (const auto val : src) {
-            const auto digits = ByteToHex(val);
+            char digits[2];
 
-            PushBack(digits.mFirst);
-            PushBack(digits.mSecond);
+            auto err = String(digits, 2).ByteToHex(val, upperCase);
+            if (!err.IsNone()) {
+                return err;
+            }
+
+            err = Insert(end(), digits, static_cast<char*>(digits) + 2);
```

Comment:
Size is tested at the start of the method. thereis no need in additional check.

Reply:
I removed checking size at the beginning then. Insert may check other errors in general case not only size.

Reply:
ok